llvm/lib/Transforms/Scalar/InferAddressSpaces.cpp

```diff
@@ -356,8 +384,14 @@ getPointerOperands(const Value &V, const DataLayout &DL,
     return {II.getArgOperand(0)};
   }
   case Instruction::IntToPtr: {
-    assert(isNoopPtrIntCastPair(&Op, DL, TTI));
-    auto *P2I = cast<Operator>(Op.getOperand(0));
+    assert(isNoopPtrIntCastPair(&Op, DL, TTI) || isZExtPtrIntCastPair(&Op, DL));
+    Operator *P2I;
+    if (isNoopPtrIntCastPair(&Op, DL, TTI)) {
+      P2I = cast<Operator>(Op.getOperand(0));
+    } else {
+      auto *ZExt = cast<Operator>(Op.getOperand(0));
+      P2I = cast<Operator>(ZExt->getOperand(0));
+    }
```

Comment:
If you return a `Value*`/`Operator*` this could be simplified
```suggestion
    Operator *P2I = getNoopOrExtPtrIntCastPair(&Op, DL, TTI)
    assert(P2I != nullptr);
```